CONTRIBUTING.rst

```diff
@@ -33,6 +33,70 @@ you are solving it. This might save you a lot of time if the maintainers
 are already working on it or have a specific idea on how the problem should
 be solved.
 
+Setting up the development environment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are three python tools required to develop, test and release ics.py:
+`poetry <https://python-poetry.org/>`_ for managing virtualenvs plus dependencies and for building plus publishing the package,
+`tox <https://tox.readthedocs.io/>`_ for running the testsuite and building the documentation,
+and `bumpversion <https://pypi.org/project/bumpversion/>`_ to help with making a release.
+Their respective configuration files are ``pyproject.toml``, ``tox.ini`` and ``.bumpversion.cfg``.
+The ``poetry.lock`` file optionally locks the dependency versions against which we want to develop,
+which is independent from the versions the library pulls in when installed as a dependency itself (where we are pretty liberal),
+and the versions we test against (which is always the latest releases installed by tox).
```

Comment:
The sentence is a bit long. :wink: Maybe you should shorten it a bit. I'm not entirely sure what you want to express with the second sentence.

```suggestion
When Poetry is finished with the installation, it locks all dependencies and their versions
in the file :file`poetry.lock`.
This is independent from the versions the library pulls in when installed as a dependency itself (where we are pretty liberal),
and the versions we test against (which is always the latest releases installed by tox).
```

Reply:
The file locks the versions of dependencies in the development environment set up by poetry, so that this environment is the same for everyone. The file is only read by poetry and not included in any distributions, thus these restrictions don't apply when running `pip install ics`. As tox manages its own environments and also doesn't read the lock file, it just installs the latest versions of dependencies for testing. More details on the poetry side can be found here:
https://python-poetry.org/docs/basic-usage/#commit-your-poetrylock-file-to-version-control

We don't necessarily have to include the file in the repo (see the link above), and I also have no strong opinion here#, I just thought that it could be nice to have for developing without any side-effects for distributions and testing.